examples/lighting-app/qpg/src/ZclCallbacks.cpp

```diff
@@ -161,12 +161,29 @@ void MatterPostAttributeChangeCallback(const chip::app::ConcreteAttributePath &
 void emberAfOnOffClusterInitCallback(EndpointId endpoint)
 {
     bool onOffValue = false;
+    EmberAfStatus status;
 
-    EmberAfStatus status = OnOff::Attributes::OnOff::Get(1, &onOffValue);
+    status = OnOff::Attributes::OnOff::Get(1, &onOffValue);
 
     if (status == EMBER_ZCL_STATUS_SUCCESS)
     {
         LightingMgr().InitiateAction(onOffValue ? LightingManager::ON_ACTION : LightingManager::OFF_ACTION, 0, 1,
                                      (uint8_t *) onOffValue);
     }
+
+    /* restore values saved by DeferredAttributePersistenceProvider */
+    app::DataModel::Nullable<uint8_t> currentLevel;
+    status = LevelControl::Attributes::CurrentLevel::Get(endpoint, currentLevel);
+    assert(status == EMBER_ZCL_STATUS_SUCCESS);
+    uint8_t * value = reinterpret_cast<uint8_t *>(&currentLevel);
```

Comment:
Why is this cast ok? I see that the InitiateAction seems to use `ChipLogProgress(NotSpecified, "LightMgr:LEVEL: lev:%u->%u", mLevel, *value)` however this is a Nullable which is a Optional, whose memory is layed out as `bool; T;` to a nullable value is not a uint8_t.